### PR TITLE
Openmpi cuda support

### PR DESCRIPTION
**Pull request by jcftang**
This PR updates the cuda package with the latest stable version of cuda (needed for the new 1080ti) and enables cuda support in openmpi.

### CODE REVIEW

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 1]
```diff
@@ -255,6 +255,14 @@ def configure_args(self):
             else:
                 config_args.append('--disable-mpi-thread-multiple')
 
+        # CUDA support
+        if spec.satisfies('@1.6:'):
+	    if '+cuda' in spec:
+                config_args.append('--with-cuda={0}'.format(spec['cuda'].prefix))
+                config_args.append('--with-cuda-libdir={0}/lib64'.format(spec['cuda'].prefix))
```
Comment (davydden) [inline var/spack/repos/builtin/packages/openmpi/package.py]: is it always `lib64`? in the past we had various issues where hardcoding `lib` or `lib64` lead to issues.
If `cuda` contains a single library named `libcuda.so` or alike, then the default interface to `libs` by @alalazo should work and you would be able to do `spec['cuda'].libs.directories` to get what you need.

Comment (davydden) [inline var/spack/repos/builtin/packages/openmpi/package.py]: look at the `LibraryList` in http://spack.readthedocs.io/en/latest/spack.html

Comment (jcftang) [inline var/spack/repos/builtin/packages/openmpi/package.py]: I'm not entirely sure if its always in lib64, cuda only has 64bit downloads for linux so I would assume that its safe to have it like this. I can make the change spec['cuda'].libs.directories

Comment (davydden) [inline var/spack/repos/builtin/packages/openmpi/package.py]: > I can make the change spec['cuda'].libs.directories

i think that would be a safe bet.

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 2]
```diff
@@ -255,6 +255,14 @@ def configure_args(self):
             else:
                 config_args.append('--disable-mpi-thread-multiple')
 
+        # CUDA support
+        if spec.satisfies('@1.6:'):
+            if '+cuda' in spec:
+                config_args.append('--with-cuda={0}'.format(spec['cuda'].prefix))
+                config_args.append('--with-cuda-libdir={0}'.format(spec['cuda'].libs.directories))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openmpi/package.py]: These lines are a bit too long and are causing the flake8 tests to fail.

**Comment by jcftang** [comment]
FYI I've tested this with openmpi 1.10.6 and 2.1.0

**Pull request merged**